text/0670-aws-applicationsignals-enablement-l2.md

```diff
@@ -0,0 +1,633 @@
+      memoryReservationMiB: 50,
+      portMappings: [{
+        containerPort: 4316,
+        hostPort: 4316,
+      }, {
+        containerPort: 2000,
+        hostPort: 2000,
+      }]
+    }
+  }
+})
+```
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `status/api-approved` label was applied to the
+RFC pull request):
+
+```text
+[x] Signed-off by API Bar Raiser @moelasmar
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We are launching a new L2 construct `ApplicationSignalsTaskDefinition` to simplify the enablement steps to onboard Application Signals for ECS customers.
+
+### Why should I use this feature?
+
+It automates the Application Signals configuration when defining the ECS TaskDefinition, eliminating complex manual setup work and reducing potential
+configuration errors.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+We're introducing this feature to improve the customer experience with Application Signals by eliminating complex manual setup requirements for ECS.
+Currently, customers face frustration due to time-consuming configuration processes and frequent setup errors, which often leads to hours spent
+troubleshooting instead of actually using the product features. By automating these setup steps, we aim to provide a smoother onboarding experience and
+enable customers to focus on utilizing the product's actual capabilities more quickly.
+
+ECS is not the only platform Application Signals supports today. On some platforms, this complaint is minor with the integration of other components,
+such as add-on on EKS, operator on native Kubernetes, and vended layer on Lambda. The pain point is most prominent on ECS and EC2.
+
+### Why should we *not* do this?
+
+We should not using this feature if granular control over the TaskDefinition configuration is required. While the automation simplifies setup, it may
+not be suitable for cases where manually customizing specific fields or maintaining precise control over the TaskDefinition settings is needed. In
+such cases, stick with the existing L1 ECS TaskDefinition construct for manual configuration.
+
+### What is the technical solution (design) of this feature?
+
+#### 1. Current enablement process
+
+At high level, the customers are required to execute the following steps to onboard Application Signals.
+
+1. Add additional CloudWatch agent permission to the Task Role.
+2. Instrument ECS [TaskDefinition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html) with 5 steps.
+    1. Configuring the bind mount volume to provide a volume for auto-instrumentation.
+    2. Adding a new non-essential CloudWatch agent sidecar.
+        1. *Not required when using CloudWatch agent as daemon on ECS on EC2.*
+    3. Adding a new non-essential `init` container with ADOT auto-instrumentation image.
```

Comment:
we agreed that this name will be auto generated to avoid conflicts. You can use this https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.Names.html#static-nodewbruniquewbridnode to generate the unique Id for this container

Reply:
Updated the description.